includes/class-wc-google-gtag-js.php

```diff
@@ -507,4 +572,22 @@ public function event_tracking_code( $parameters, $selector ) {
 		);
 	}
 
+	/**
+	 * Enqueue inline Javascript to trigger a custom jQuery event
+	 *
+	 * @param string $event_name Name of the event which will be prefixed with `google_analytics-`
+	 * @param array  $event_data Data to include with the event
+	 *
+	 * @return void
+	 */
+	public function trigger_event( $event_name, $event_data ) {
+		wc_enqueue_js(
+			sprintf(
+				"$( document.body ).trigger( 'google_analytics-%s', '%s' )",
+				esc_js( $event_name ),
+				wp_json_encode( $event_data )
```

Comment:
Also, can we call the `trackAddToCart` function synchronously? Instead of bubbling events?

I can imagine we can do so with
1. publishing the extension's namespace in the global scope for some shared stuff
```php
				"window['woocommerce-google-analytics-integration'].trackEvent('%s', '%s' )",
				esc_js( $event_name ),
				wp_json_encode( $event_data )
```
2. attaching the extension's shared namespace to its own script tag. Script tag, whose handle we control
```php
				"document.getElementById('%s').wcgai.trackEvent('%s', '%s' )",
				$this->script_handle . '-actions-js',
				esc_js( $event_name ),
				wp_json_encode( $event_data )
				
				// in actions.js
				document.currentScript.wcgai = { //...
```

Each solution - yours included - relies on squatting a unique name in the public, global space. So I'd say each is equally bad in scope pollution metric. So we should evaluate other aspects :;)
1. We introduce a global variable - which makes things least magical and easy to inspect
2. We rely on the handle id that needs to be there for other reasons. So, theoretically, we do not pollute more than before. However, it's slightly tricky the way we access the scope.
3. Event-based solution. Relies on a unique event name. Not so straightforward and inspectable. For example. I imagine we can face a racy condition when we fire an event before registering the listener, then it would be hard to notice that something went wrong. Logs will be clean for HEs to check.